### PR TITLE
Migrate Docs to Docusaurus

### PR DESCRIPTION
**Pull request by ifestus**
**Summary:** Simple configuration and organization changes/additions to support publishing to GitHub Pages with Docusaurus. There are also a few changes to the scripts used for publishing in the CI pipeline.

Addresses [CUMULUS-956: Migrate developer documentation and data-cookbooks to docusaurus](https://bugs.earthdata.nasa.gov/browse/CUMULUS-956)

## Changes

* removed gitbook-cli references and dependencies
* added headers (wrapped in `---`) at the beginning of all `.md` docs
* moved images into `docs/assets`
* new `website/` directory at the root level of the repository - this is where website configuration, building, and publishing is done
* `website/siteConfig.js` is where most of the configuration lies. This is where the navigation bar, url, color-schemes, titles, and other settings exist. [Documentation](https://docusaurus.io/docs/en/site-config)
* `website/sidebars.json` is the configuration for sidebar(s). [Documentation](https://docusaurus.io/docs/en/navigation.html)
* small changes to `travis-ci/deploy-to-github-pages.sh` to build docs with Docusaurus

## Manual Testing
1. Pull this branch (`docusaurus-proto`)
2. run `(cd website && yarn install) && yarn docs-start` from the repo's root directory - this will start a local server for your viewing pleasure.

## Test Plan
Things that should succeed before merging.

- [ ] Update CHANGELOG
- [ ] Manual Testing

### CODE REVIEW

**Comment by ifestus** [comment]
Just as a note - we're still working on `Search` functionality. It requires a very simple setup on Algolia's DocSearch website and then 2 configuration changes in `website/siteConfig.js`

File: website/siteConfig.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    //caption: 'User1',
+    // You will need to prepend the image path with your baseUrl
+    // if it is not '/', like: '/test-site/img/docusaurus.svg'.
+    //image: '/img/docusaurus.svg',
+    //infoLink: 'https://www.facebook.com',
+    //pinned: true,
+  }
+];
+
+const siteConfig = {
```
Comment (ifestus) [inline website/siteConfig.js]: Configuration for the website.

File: website/sidebars.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "docs": {
```
Comment (ifestus) [inline website/sidebars.json]: Configuration of the sidebars.

**Comment by ifestus** [comment]
@laurenfrederick good question. I'll look at adding some documentation for adding documentation and explaining how to test locally. The Docusaurus readme _might_ have some useful information, but it's general Docusaurus information (definitely not specific to Cumulus).

Running locally should be a simple two steps:
* checkout `docusaurus-proto`
* run `(cd website && yarn install) && docs-start` from the cumulus root directory

The above steps should start a local server that you can connect to and test against.

File: example/README.md [hunk 3]
```diff
@@ -85,3 +85,34 @@ your stack name [here](../travis-ci/select-stack.js).
 There are tests for redeploying the Cumulus stack while a workflow is running (see `spec/redeployment`). This is acheived by backing up with the `workflows.yml` file, updating it, and redeploying the stack. When redeploy tests are complete, the original `workflows.yml` is restored, the backup file is deleted, and the stack is redeployed, restoring it to its original state.
 
 Please note that the stack will be redeployed multiple times when running tests and any errors during redeployment can result in errors in later tests. The deployment output is printed to the console.
+
+## Cumulus Documentation
```
Comment (laurenfrederick) [inline example/README.md]: This is in `example/README` and should go in the `README` in the top level cumulus directory.

File: docs/doc_installation.md [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+title: Title Of Doc  # Whatever title you feel like adding. This will show up as the index to this page on the sidebar.
+hide_title: true     # So the title of the Doc doesn't show up at the top of the webpage (generally we already have the title written as h1 in the documentation).
+---
+```
+
+**Note:** To have the new page show up in a sidebar the designated `id` must be added to a sidebar in the `website/sidebars.js` file. Docusaurus has an in depth explanation of sidebars [here](https://docusaurus.io/docs/en/navigation).
+
+### Versioning Docs
+
+We lean heavily on Docusaurus for versioning. Their suggestions and walkthrough can be found [here](https://docusaurus.io/docs/en/versioning). It is worth noting that we would like the Documentation versions to match up directly with release versions. This version naming/cutting is explained in the Cumulus repo's top level [README.md](https://github.com/nasa/cumulus/blob/master/README.md#5-cut-new-version-of-cumulus-documentation).
```
Comment (ifestus) [inline docs/doc_installation.md]: The repo link to `README.md` won't work until this branch has been merged into master.

**Comment by ifestus** [comment]
Another note from the gotomeeting discussion - I'm currently waiting for API keys from [DocSearch](https://community.algolia.com/docsearch/) so I can update the `siteConfig.js` to support searching.

File: docs/doc_installation.md (hunk 4 above)
Comment (laurenfrederick) [reply docs/doc_installation.md]: 👍

**Pull request merged**